org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/AdapterUtils.java

```diff
@@ -0,0 +1,325 @@
+                }
+            }
+        };
+        SearchEngine searchEngine = new SearchEngine();
+        searchEngine.search(
+                pattern,
+                new SearchParticipant[] { SearchEngine.getDefaultSearchParticipant() },
+                scope,
+                requestor,
+                null /* progress monitor */);
+
+        return projects;
+    }
+
+    /**
+     * Gets java project from projectName and main class.
+     * @param projectName
+     *                   project name
+     * @param mainClass
+     *                   full qualified class name
+     * @return java project
+     * @throws CoreException
+     *                      CoreException
+     */
+    public static IJavaProject getJavaProject(String projectName, String mainClass) throws CoreException {
+        // if type exists in multiple projects, debug configuration need provide project name.
+        if (projectName != null) {
+            return getJavaProjectFromName(projectName);
+        } else {
+            List<IJavaProject> projects = AdapterUtils.getJavaProjectFromType(mainClass);
+            if (projects.size() == 0 || projects.size() > 1) {
+                throw new CoreException(new Status(IStatus.ERROR, JavaDebuggerServerPlugin.PLUGIN_ID, "project count is zero or more than one."));
+            }
+            return projects.get(0);
+        }
+    }
+
+    /**
+     * Accord to project name and main class, compute runtime classpath.
+     * @param projectName
+     *                   project name
+     * @param mainClass
+     *                   full qualified class name
+     * @return class path
+     * @throws CoreException
+     *                      CoreException
+     */
+    public static String computeClassPath(String projectName, String mainClass) throws CoreException {
+        IJavaProject project = getJavaProject(projectName, mainClass);
+        return computeClassPath(project);
+    }
+    
+    /**
+     * Compute runtime classpath.
+     * 
+     * @param javaProject
+     *            java project
+     * @return class path
+     * @throws CoreException
+     *             CoreException
+     */
+    public static String computeClassPath(IJavaProject javaProject) throws CoreException {
+        if (javaProject == null) {
+            throw new IllegalArgumentException("javaProject is null");
+        }
+        String[] classPathArray = JavaRuntime.computeDefaultRuntimeClassPath(javaProject);
+        String classPath = String.join(System.getProperty("path.separator"), classPathArray);
+        return classPath;
+    }
+
+    /**
+     * Searches the source file from the full qualified name and returns the uri string.
+     * @param project
+     *               project instance
+     * @param fqcn
+     *               full qualified name
+     * @return the uri string of source file
+     */
+    public static String getURI(IProject project, String fqcn) {
```

Comment:
For the parameter in API  "fqcn", should avoid use the abbreviation format in order to provide a clean API interface.